modules/ShowersMatcher.py

```
@@ -0,0 +1,427 @@
+    #                     numerator = intersection_matrix[idx_f, idx_t]
+    #                     denominator1 = truth_sum_matrix[0, idx_t]
+    #                     denominator2 = pred_sum_matrix[idx_f, 0]
+    #                     score = numerator / min(denominator1, denominator2)
+    #                 if score > self.iou_threshold:
+    #                     C[i, j] = score
+    #
+    #         # # Let's match these guys again
+    #         C = np.array(C)
+    #         row_id, col_id = linear_sum_assignment(C, maximize=True)
+    #         for r, c in zip(row_id, col_id):
+    #             if C[r, c] > 0:
+    #                 p = pairs[r]
+    #                 if matched_full_graph.nodes(data=True)[free_nodes[c]]['type'] == NODE_TYPE_TRUTH_SHOWER:
+    #                     # Free node was a truth node
+    #                     matched_full_graph.add_edge(next(iter(p[0])), free_nodes[c], attached_in_pass=npass)
+    #                 else:
+    #                     # Free node was a pred node
+    #                     matched_full_graph.add_edge(next(iter(p[1])), free_nodes[c], attached_in_pass=npass)
+    #         graphs.append(matched_full_graph.copy())
+    #
+    #
+    #     self.calculated_graph = self._reduce_graph(matched_full_graph)
+
+
+    def process(self):
+        self._build_data_graph()
+
+        if self.match_mode == 'iou_max':
+            self._match_single_pass()
+        elif self.match_mode == 'iom_max_multi':
+            raise NotImplementedError('Multi passing needs to be re-written')
+            # self._match_multipass()
+        elif self.match_mode == 'emax_iou':
+            raise NotImplementedError('Need to finish some work')
+            # self._match_single_pass()
+        elif self.match_mode == 'emax_angle':
+            raise NotImplementedError('Need to finish some work')
+            # self._match_single_pass()
+
+    def get_hit_data(self):
```

maybe throw an exception until implemented?